### PR TITLE
docs build - publish `push`-based docsite to GitHub pages

### PR DESCRIPTION
**Pull request by briantist**
##### SUMMARY

cc @tremble (discussed in IRC)

This collection is already using the docs build workflows, what is proposed here is to enable GitHub pages on the repository, and publish rendered versions of the docs on `push`, for the `main` and `stable-*` branches (and tags).
This replaces the manually generated RST files in docs.

We've done the same so far on `community.windows` and `ansible.windows`, see:
- https://github.com/ansible-collections/ansible.windows/pull/388

**Before merging, GitHub Pages must be enabled by someone with the permission to do so.**
Instructions are available here:
- https://github.com/ansible-community/github-docs-build/wiki/Getting-started-with-GitHub-Pages#setting-up-github-pages-for-the-first-time

If this works out a similar PR can be put up on `amazon.aws`, by me or a collection maintainer.

---

Additional things this enables (via `docsite/links.yml`):
* "Edit on GitHub" button automatically added to docs pages
* Communications links to the matrix and IRC rooms

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,211 +35,7 @@ All support for the original AWS SDK `boto` was removed in release 4.0.0.
...
-[community.aws.wafv2_resources](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_resources_module.rst)|wafv2_web_acl
-[community.aws.wafv2_resources_info](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_resources_info_module.rst)|wafv2_resources_info
-[community.aws.wafv2_rule_group](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_rule_group_module.rst)|wafv2_web_acl
-[community.aws.wafv2_rule_group_info](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_rule_group_info_module.rst)|wafv2_web_acl_info
-[community.aws.wafv2_web_acl](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_web_acl_module.rst)|Create and delete WAF Web ACLs
-[community.aws.wafv2_web_acl_info](https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.wafv2_web_acl_info_module.rst)|wafv2_web_acl
+See the complete list of collection content in the [Plugin Index](https://ansible-collections.github.io/community.aws/branch/main/collections/ansible/windows/index.html#plugin-index).
```
Comment (tremble) [inline README.md:38]:
```suggestion
See the complete list of collection content in the [Plugin Index](https://ansible-collections.github.io/community.aws/branch/main/collections/community/aws/index.html#plugin-index).
```

I'm assuming this is where the GitHub pages land?

**Review by tremble — COMMENTED** [review]
I'm very much +1 to concept behind this, although I think there's a typo in the README

@markuman @alinabuzachis @jillr ?

**Review by tremble — APPROVED** [review]

**Comment by jillr** [comment]
I think I'm just out of the loop - deleting all the RST files here will not impact the docs that get published to docs.ansible.com, is that correct?  This just removes the need to manually run add_docs.py before a release and allows the GH Action to generate the docs that go on docs.a.c for us automatically?

**Comment by briantist** [comment]
> I think I'm just out of the loop - deleting all the RST files here will not impact the docs that get published to docs.ansible.com, is that correct? This just removes the need to manually run add_docs.py before a release and allows the GH Action to generate the docs that go on docs.a.c for us automatically?

@jillr That's correct! The existence or absence of those RST files does not affect anything that goes on the public docsite.

The only thing here that affects that is the addition of `docs/docstie/links.yml`; that change would get reflected on the public docsite regardless of any other changes in this PR.

**Comment by briantist** [comment]
@jillr also if it's helpful, here's what the "local" docsites look like for `community.windows`:
- https://ansible-collections.github.io/community.windows/branch/main/
- https://ansible-collections.github.io/community.windows/tag/1.11.0/

And their README change (they used to do the manual RST generating and linking step as well):
- https://github.com/ansible-collections/community.windows/#included-content

**Comment by jillr** [comment]
@briantist thanks for explaining!  I'll get GH pages enabled on both AWS collection repos tomorrow

**Comment by tremble** [comment]
@briantist Do you happen to know if there's an easy way for us to trigger docs builds for the existing tags?  I can see this being helpful.

**Comment by jillr** [comment]
@briantist Do we need to create a gh-pages branch?  (I'm looking at the Pages config on the windows collections).  Pages is now configured on this repo, set to build from branch `main`.

**Comment by briantist** [comment]
> @briantist Do we need to create a gh-pages branch? (I'm looking at the Pages config on the windows collections). Pages is now configured on this repo, set to build from branch `main`.

@jillr 
Ah, I think that will have to be reset. At the moment these workflows require the setup to use `gh-pages`, instructions are here:
https://github.com/ansible-community/github-docs-build/wiki/Getting-started-with-GitHub-Pages#setting-up-github-pages-for-the-first-time

**Comment by jillr** [comment]
@briantist You did give me that link, apologies and thank you for humoring me.  :grin:  This repo should be all set!

**Comment by samccann** [comment]
@tremble - I don't know if this line ever worked for this collection, but the docsite link should be updated:
https://github.com/ansible-collections/community.aws/blob/main/galaxy.yml#L13

**Comment by briantist** [comment]
> @tremble - I don't know if this line ever worked for this collection, but the docsite link should be updated: https://github.com/ansible-collections/community.aws/blob/main/galaxy.yml#L13

It did "work" in that the directory contained RST files. But that's a good catch, it should be updated. The most analogous choice now is the GH pages site, but I would argue that link should probably point to the devel docs on docs.ansible.com, since that link will be what galaxy users will see, and the devel docs will correspond to the latest version in galaxy.

**Pull request merged**